src/main.rs

```diff
@@ -6,17 +13,188 @@ fn show_info() {
+
+        // Initialize an output block with zeros
+        let mut output_block = [0.0; 2];
+
+        // Process the input block
+        fir_comb_filter.process(&[&input_block], &mut [&mut output_block]);
+
+        // Assert that the output is zero
+        assert_eq!(output_block, [0.0; 2]);
+    }
+    
+
+    #[test]
+    fn test_iir_magnitude_change_if_input_freq_matches_feedback() {
+        let max_delay_secs = 0.1;
+        let sample_rate_hz = 44100.0;
+        let num_channels = 1;
+        let feedforward_coeff = 0.0;
+        let feedback_coeff = 0.5;
+        let mut iir_comb_filter = IIRCombFilter::new(max_delay_secs, sample_rate_hz, num_channels, feedforward_coeff, feedback_coeff).unwrap();
+        let input_block = [1.0; 1024];
+        let mut output_block = [0.0; 1024];
+        iir_comb_filter.process(&[&input_block], &mut [&mut output_block]);
+        // Check if magnitude has changed (assuming the input block has non-zero magnitude)
+        assert_ne!(output_block, [0.0; 1024]);
+    }
+
+    #[test]
+    fn test_fir_iir_correct_result_for_varying_input_block_size() {
+        let max_delay_secs = 0.1;
+        let sample_rate_hz = 44100.0;
+        let num_channels = 1;
+        let feedforward_coeff = 0.5;
+        let feedback_coeff = 0.0;
+        let mut fir_comb_filter = FIRCombFilter::new(max_delay_secs, sample_rate_hz, num_channels, vec![feedforward_coeff, -feedforward_coeff]).unwrap();
+        let mut iir_comb_filter = IIRCombFilter::new(max_delay_secs, sample_rate_hz, num_channels, feedforward_coeff, feedback_coeff).unwrap();
+        
+        let input_block_sizes = [256, 512, 1024, 2048];
+        for &block_size in input_block_sizes.iter() {
+            let input_block = vec![1.0; block_size];
+            let mut output_block = vec![0.0; block_size];
+            fir_comb_filter.process(&[&input_block], &mut [&mut output_block]);
+            iir_comb_filter.process(&[&input_block], &mut [&mut output_block]);
+            // Assert whatever properties you want to test for each block size
+        }
+    }
+
+    #[test]
+    fn test_fir_iir_correct_processing_for_zero_input_signal() {
+        let max_delay_secs = 0.1;
+        let sample_rate_hz = 44100.0;
+        let num_channels = 1;
+        let feedforward_coeff = 0.5;
+        let feedback_coeff = 0.0;
+        let mut fir_comb_filter = FIRCombFilter::new(max_delay_secs, sample_rate_hz, num_channels, vec![feedforward_coeff, -feedforward_coeff]).unwrap();
+        let mut iir_comb_filter = IIRCombFilter::new(max_delay_secs, sample_rate_hz, num_channels, feedforward_coeff, feedback_coeff).unwrap();
+        
+        let input_block = vec![0.0; 1024];
+        let mut output_block = vec![0.0; 1024];
+        fir_comb_filter.process(&[&input_block], &mut [&mut output_block]);
+        iir_comb_filter.process(&[&input_block], &mut [&mut output_block]);
+        // Assert whatever properties you want to test for zero input signal
+    }
```

Comment:
You could assert that the signal is still full of zeros to ensure your silence remains silent after processing.